src/local-git/GitLocalInterface.ts

```diff
@@ -28,18 +29,18 @@ export class GitLocal implements NavigatorBackend {
   }
 
   //Actions: push to the origin repo
-  async pushCommitstoRepo(branchName: string, repoPath: string) {
+  pushCommitstoRepo(branchName: string, repoPath: string) {
     let repo: nodegit.Repository, remote: nodegit.Remote;
     //Local repo
-    await nodegit.Repository.open(repoPath)
+    nodegit.Repository.open(repoPath)
```

Comment:
I think you'll need to return this promise so that `await pushCommitstoRepo` actually waits for this promise to finish

```suggestion
    return nodegit.Repository.open(repoPath)
```